pex/pex_rules.bzl

```diff
@@ -82,6 +83,16 @@ def _collect_transitive_reqs(ctx):
   return transitive_reqs
 
 
+def _collect_repos(ctx):
+  repos = {}
+  for dep in ctx.attr.deps:
+    if hasattr(dep.py, "repos"):
+      repos += dep.py.repos
+  for file in egg_file_types.filter(ctx.files.repos):
```

Comment:
Would it work to allow source distributions in `repos` along with eggs and wheels? That could be very convenient at times.